network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.kalium.network.exceptions
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+// TODO: return custom error instead of throwing exception
+
+@Serializable
+sealed class KaliumException : Exception()
+
+@Serializable
+sealed class NetworkException(
+    open var _message: String? = null,
+    open var _code: Int = 0,
+    open var _label: String? = null
+) : KaliumException()
+
+@Serializable
+data class HttpException(
+    @SerialName("message") override var _message: String? = null,
+    @SerialName("code") override var _code: Int = 0,
+    @SerialName("label") override var _label: String? = null,
+
+    ) : NetworkException(_message) {
+
+
+    constructor(
+        message: String?,
+        code: Int
+    ) : this(_message = message) {
+        this._code = code
+
+        this._message = message
+    }
+
+
+    override fun toString(): String {
+        val clazz = this::class.simpleName
+        return "$clazz: code: $_code, msg: $message, label: $_label"
+    }
+}
+
+@Serializable
+class AuthException : NetworkException {
+    constructor(message: String?, code: Int) : super(_message = message, _code = code)
+    constructor(code: Int) : super(_code = code)
+
+    constructor(
+        message: String?,
```

Comment:
Auth and HttpException are not needed now (this was part of the orginal java code)

Reply:
`AuthException` is being used by `AuthApiImp`. It's a bit out of the scope of this PR refactoring and changing the logic inside the `network` module.